manifests/rule.pp

```diff
@@ -162,7 +162,7 @@
     $filename = "${ferm::configdirectory}/chains/${table}-${chain}.conf"
   }
 
-  $rule = squeeze("${comment_real} ${proto_real} ${proto_options_real} ${dport_real} ${sport_real} ${daddr_real} ${saddr_real}${outerface_real} ${action_real}${to_source_real};", ' ')
+  $rule = join([strip(squeeze("${comment_real} ${proto_real} ${proto_options_real} ${dport_real} ${sport_real} ${daddr_real} ${saddr_real} ${outerface_real} ${action_real} ${to_source_real}", ' ')), ';'])
```

Comment:
Why do we now suddenly need `join()` and `strip()`?
From what I can tell you're only adding strings and everyting is inside a single string interpolation anyway.

Reply:
Because `to ...` needs to be _after_ `SNAT`, and `<space>${to_source_real}` results in a single space between `${action_real}` and `;` in all rules that have an empty `${to_source_real}`.

Reply:
That extra space makes all (other) acceptance tests fail.

Reply:
I could make it more clear using a `regsubst()` maybe.

Reply:
Can you provide compiled/expanded results for each?
I.e. output for the version that doesn't work and output for the one that does work. I'm having a little trouble seeing why any of that `regsubst()` or similar is needed.

Reply:
Here's an example of why tests fail when a rule ends with `<space><semi colon>`:

```
  84) ferm::rule on centos-7-x86_64  definining rules in custom chains is expected to contain Concat::Fragment[SSH-allow-ssh-localhost] with content  supplied string
      Failure/Error: it { is_expected.to contain_concat__fragment('SSH-allow-ssh-localhost').with_content("mod comment comment 'allow-ssh-localhost' proto tcp dport 22 saddr @ipfilter((127.0.0.1)) ACCEPT;\n") }
      
        expected that the catalogue would contain Concat::Fragment[SSH-allow-ssh-localhost] with content set to supplied string
        Diff:
        @@ -1,2 +1,2 @@
        -mod comment comment 'allow-ssh-localhost' proto tcp dport 22 saddr @ipfilter((127.0.0.1)) ACCEPT;
        +mod comment comment 'allow-ssh-localhost' proto tcp dport 22 saddr @ipfilter((127.0.0.1)) ACCEPT ;
```